test/TBD/multimodule-implied-objc.swift

```diff
@@ -0,0 +1,47 @@
+// REQUIRES: VENDOR=apple
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/cache)
+// RUN: split-file %s %t
+
+/// Create Swift modules to import.
+// RUN: %target-swift-emit-module-interface(%t/IndirectMixedDependency.swiftinterface) \
+// RUN:   %t/IndirectMixedDependency.swift -I %t
+// RUN: %target-swift-emit-module-interface(%t/SwiftDependency.swiftinterface) \
+// RUN:   %t/SwiftDependency.swift -I %t
+
+// Generate TBD file.
+// RUN: %target-swift-frontend %t/Client.swift -emit-ir -o/dev/null \
+// RUN: -parse-as-library -module-name test -validate-tbd-against-ir=missing \
+// RUN: -emit-tbd -emit-tbd-path %t/client.tbd \
+// RUN:   -I %t -module-cache-path %t/cache 2>&1 
+
+// RUN: %FileCheck %s < %t/client.tbd
+// CHECK: objc-classes: [ _TtCO4test11extendedAPI6Square ]
+// CHECK-NOT: _OBJC_CLASS_$
+// CHECK-NOT: _OBJC_METACLASS_$
+
+//--- module.modulemap
+module IndirectMixedDependency {
+  header "IndirectMixedDependency.h"
+}
+
+//--- IndirectMixedDependency.h
+#import <Foundation/Foundation.h>
```

Comment:
Depending on the SDK's Foundation may slow down the test or lead to surprises. It's a good idea to use the mock-sdk `// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) ...` if you can.